todo/commands/settings.py

```diff
@@ -225,6 +235,57 @@ async def todo_autosort(self, ctx: commands.Context, value: bool):
         await self.cache.set_user_setting(ctx.author, "autosorting", value)
         await ctx.send(f"Autosorting is now {enabled}")
 
+    @commands.check(no_markdown)
+    @todo_settings.command(name="todoemoji", aliases=("temoji",))
+    async def todo_emoji(self, ctx: commands.Context, reset: bool = False):
+        """Set the emoji used for todos
+
+        This will prompt you to react with an emoji. Note that the emoji must be one the bot can use.
+
+        **Arguments**
+            - `reset` Whether to reset the emoji back to default.
+        """
+
+        if reset:
+            await self.cache.set_user_setting(ctx.author, "todo_emoji", None)
+            return await ctx.send("Done. Your emoji has been reset.")
+        msg = await ctx.send("Please react with the emoji you want for the uncompleted todos")
+        pred = ReactionPredicate.same_context(message=msg, user=ctx.author)
+        try:
+            emoji = (await self.bot.wait_for("reaction_add", check=pred))[0].emoji
+        except asyncio.TimeoutError:
+            return await ctx.send("Okay, I won't set your emojis")
+        if not isinstance(emoji, str) and not self.bot.get_emoji(emoji.id):
```

Comment:
Unless I'm missing something here, how can a string have an `id` attribute?

Same applies for L284

Reply:
First `not` check guards against the emoji being a string, not the entire `if` statement